### ISSUE TITLE
0.20.5 UMD module loaded via github does not set default import correctly

### ISSUE THREAD

**Issue by ghempton** [systemjs/systemjs#1563]
This previously worked without issue before the 0.20 upgrade. Here is a complete example of the issue: https://github.com/ghempton/systemjs-builder-test/tree/cjs-github

I have tried explicitly setting the format to `cjs`, `amd`, etc. but in all cases it seems to set the default to be the module itself. E.g. if I have `import sizeMe from 'react-sizeme'` where sizeMe is the module in question, when the module executes, `sizeMe` is the module itself whereas it is expected that it should be `sizeMe.default`.

```javascript
System.register(['react', 'react-dom', 'react-sizeme'], function (_export, _context) {
  "use strict";

  var React, ReactDOM, sizeMe;
  return {
    setters: [function (_react) {
      React = _react.default;
    }, function (_reactDom) {
      ReactDOM = _reactDom.default;
    }, function (_reactSizeMe) {
      sizeMe = _reactSizeMe.default;
    }],
    execute: function () {
      // sizeMe should be the function but is instead the module. Need to access `sizeMe.default` to get the expected function

    }
  };
});

```

**Comment by asapach** [comment]
My guess is that the dist version of `react-sizeme` doesn't set `__esModule` property, so it's interpreted as legacy `cjs`. Try setting the new `esModule` meta config: https://github.com/systemjs/systemjs/blob/master/docs/config-api.md#meta

**Comment by ghempton** [comment]
I just tried setting `esModule` via met config to no avail. Would this have any impact on default exports?

**Comment by asapach** [comment]
Yes, I think it's a consequence of the first breaking change mentioned here: https://github.com/systemjs/systemjs/releases/tag/0.20.0

**Comment by ghempton** [comment]
That seems to only refer to named exports. In this case I am just trying to get access to the default.

**Comment by guybedford** [comment]
The issue here was that the module is being defined as an AMD module of the form:

```javascript
{
  __esModule: true,
  default: ...
}
```

Which is then interpreted as a module in SystemJS as:

```javascript
{
  __useDefault: true,
  default: {
    __esModule: true,
    default: {...}
  }
}
```

We then apply namespace lifting, which copies exports from `default` to the base-level module, but *excluding the `default` export*.

I've adjusted the namespace lifting process to rather be a full namespace creation including the default export in https://github.com/ModuleLoader/es-module-loader/commit/f2bb1395231e017594dd2e2ff94af5086a92c854.

It will be a little while for this to release though as it needs to be tested quite widely first.

**Comment by ghempton** [comment]
I just tested this against a custom SystemJS build against the latest es-module-loader. I can verify that this did fix this issue, but also introduced a new regression: https://github.com/ModuleLoader/es-module-loader/issues/526

**Comment by guybedford** [comment]
Released in 0.20.7.

**Comment by pocesar** [comment]
this fixed a problem with `redux-thunk`, the `{default: {default {}}` problem. just need to make sure, when using jspm, to use `jspm dl-loader --latest`

**Comment by guybedford** [comment]
Unfortunately the fix here has resulted in https://github.com/systemjs/systemjs/issues/1629, so this work will have to be reverted after all.

I think this case is a more valid output than the above break, so this will have to be a known issue with the interop process.

It does make sense actually, since if a module exports `__esModule` as well as a `default` export, the module you get when importing it should be exactly that module with both an `__esModule` and `default` export. I don't think there should be any lifting in this case.

**Comment by guybedford** [comment]
Reverted as described in https://github.com/systemjs/systemjs/issues/1629#issuecomment-290369791.

**Comment by guybedford** [comment]
The reversion here has been released in 0.20.11 (https://github.com/systemjs/systemjs/releases/tag/0.20.11).
